common/tier4_state_rviz_plugin/src/autoware_state_panel.cpp

```diff
@@ -298,7 +300,7 @@ void AutowareStatePanel::onClickAutowareEngage()
   }
 
   client_engage_->async_send_request(
-    req, [this](rclcpp::Client<tier4_external_api_msgs::srv::Engage>::SharedFuture result) {
+    req, [this](rclcpp::Client<Engage>::SharedFuture result) {  // NOLINT
```

Comment:
Why NOLINT?

Reply:
The Warning message is `warning: the const qualified parameter 'result' is copied for each invocation; consider making it a reference [performance-unnecessary-value-param]`.
When I follow it and modify like `const rclcpp::Client<Engage>::SharedFuture &result`, the another error occurs.

Reply:
The error is below.
```
--- stderr: tier4_state_rviz_plugin                                 
/home/hrkota/pilot-auto.s1/src/autoware/universe/common/tier4_state_rviz_plugin/src/autoware_state_panel.cpp: In member function ‘void rviz_plugins::AutowareStatePanel::onClickEmergencyButton()’:
/home/hrkota/pilot-auto.s1/src/autoware/universe/common/tier4_state_rviz_plugin/src/autoware_state_panel.cpp:329:6: error: no matching function for call to ‘rclcpp::Client<tier4_external_api_msgs::srv::SetEmergency>::async_send_request(std::shared_ptr<tier4_external_api_msgs::srv::SetEmergency_Request_<std::allocator<void> > >&, rviz_plugins::AutowareStatePanel::onClickEmergencyButton()::<lambda(const SharedFuture&)>)’
  329 |     });

```

Reply:
I see. The official ROS 2 also uses copy, so it's okay.
https://docs.ros.org/en/rolling/Tutorials/Intermediate/Tf2/Writing-A-Tf2-Listener-Cpp.html?highlight=sharedfuture

But could update this config instead of using `NOLINT`?
https://github.com/autowarefoundation/autoware/blob/4493f0915368b534f8f1926a39fa9a101bb8eef5/.clang-tidy#L420-L421

Reply:
It's a semicolon-separated list.
https://releases.llvm.org/10.0.0/tools/clang/tools/extra/docs/clang-tidy/checks/performance-unnecessary-value-param.html

Reply:
Should I add only `SharedFuture` ? Or add `Shared.*`?
`SharedFuture` is defined here and the other Shared*** also exists.
https://github.com/boschresearch/ros2_rclcpp/blob/cbg-executor-foxy/rclcpp/include/rclcpp/client.hpp#L181-L191

Reply:
Hmm, I think adding only `SharedFuture` is okay for now.

Reply:
Or if you can find other types used in Autoware, you can check the clang-tidy behavior and add them.

Reply:
I created https://github.com/autowarefoundation/autoware/pull/2794

Reply:
@h-ohta Thanks! :pray: Have you confirmed the behavior?

Reply:
Now checking..

Reply:
@kenji-miyake In my environment, there are no errors with https://github.com/autowarefoundation/autoware/pull/2794